dependency-injection/README.md

```diff
@@ -49,3 +49,74 @@ better readability.
 > When registering a generic factory function, make sure you use
 > `HttpContext.RequestServices` instead of root service provider. Otherwise
 > scoped dependencies will cause a runtime error.
+
+## Use Keyed DI Services Service Container
+
+Keyed Services allows us to register these services with a key for the service
+implementations registered with the same service type for our ease of use and to
+access them with this key when calling them.
+
+In the list below you can see extensions that you can use when
+registering services with a key.
+
+- `AddKeyedTransient<TService, TImplementation>(object key)`
+- `AddKeyedScoped<TService, TImplementation>(object key)`
+- `AddKeyedSingleton<TService, TImplementation>(object key)`
+
+As an example, let's assume that we have an `IPersonal` interface and two
+`PersonalA` and `PersonalB` services derived from it and register them as
+singletons. Let's use an enum to make it easy to recognize.
+
+```csharp
+enum ServiceImplementation
+{
+    PersonalA,
+    PersonalB
+}
+
+builder.Services.AddKeyedSingleton<IPersonal, PersonalA>(ServiceImplementation.PersonalA);
+builder.Services.AddKeyedSingleton<IPersonal, PersonalB>(ServiceImplementation.PersonalB);
+```
+
+Now that we have registered them, we need to figure out how to call these
+services. We can use many ways for this. Some of the suggested ways:
+
+- `FromKeyedServices` attribute
+- `GetRequiredKeyedService` extension of `IServiceProvider`
+
+With the above case in mind, below are examples of the use of these pathways.
+
+```csharp
+public void FromKeyedServicesUsing([FromKeyedServices(ServiceImplementation.PersonalA)] IPersonal personal)
+{
+    ...
+}
+
+public void GetRequiredKeyedServiceUsing()
+{
+    var personal = _serviceProvider.GetRequiredKeyedService<IPersonal>(ServiceImplementation.PersonalB);
+    ...
+}
```

Comment:
do NOT repeat microsoft documentatoin and what you already demonstrate in code. keep the documentation minimal